extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/theta/SketchAggregatorFactory.java

```diff
@@ -104,20 +99,37 @@ public AggregatorAndSize factorizeWithSize(ColumnSelectorFactory metricFactory)
   @Override
   public BufferAggregator factorizeBuffered(ColumnSelectorFactory metricFactory)
   {
-    ColumnCapabilities capabilities = metricFactory.getColumnCapabilities(fieldName);
-    if (capabilities != null && capabilities.isArray()) {
-      throw InvalidInput.exception("ARRAY types are not supported for theta sketch");
-    }
+    validateInputs(metricFactory.getColumnCapabilities(fieldName));
     BaseObjectColumnValueSelector selector = metricFactory.makeColumnValueSelector(fieldName);
     return new SketchBufferAggregator(selector, size, getMaxIntermediateSizeWithNulls());
   }
 
   @Override
   public VectorAggregator factorizeVector(VectorColumnSelectorFactory selectorFactory)
   {
+    validateInputs(selectorFactory.getColumnCapabilities(fieldName));
     return new SketchVectorAggregator(selectorFactory, fieldName, size, getMaxIntermediateSizeWithNulls());
   }
 
+  private void validateInputs(@Nullable ColumnCapabilities capabilities)
+  {
+    if (capabilities != null) {
+      if (capabilities.isArray() || (capabilities.is(ValueType.COMPLEX) && !(
+          SketchModule.THETA_SKETCH_TYPE.equals(capabilities.toColumnType()) ||
+          SketchModule.MERGE_TYPE.equals(capabilities.toColumnType()) ||
+          SketchModule.BUILD_TYPE.equals(capabilities.toColumnType())))
+      ) {
+        throw new ISE(
+            "Invalid input [%s] of type [%s] for [%s] aggregator [%s]",
```

Comment:
Checkout extrapolations in the error message style guide. Third extrapolation is incorrect syntactically. 
Also, the error message isn't correct, especially the third one. Aggregator doesn't have a type  if I am not mistaken. We can simply use 
```suggestion
            "Invalid input [%s] of type [%s] for aggregator [%s]",
```
Also, let's reword it as "Not supported", instead of invalid input.